rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/CompactView.java

```diff
@@ -655,17 +511,22 @@ public void setSelectedTile(boolean selected) {
         }
 
         private void highlightGroup(boolean selected) {
-            // highlight every node following this one in the flow layout,
+            // highlight every node following this one in the normalFlow layout,
             // until we hit one from another host/ns
             if (host != null || nodesource != null) {
-                int id = CompactView.this.curTiles.indexOf((host == null) ? nodesource.getSourceName() : host.getId());
+                int id;
+                if (host == null) {
+                    id = CompactView.this.normalFlow.indexOf(nodesource).get();
+                } else {
+                    id = CompactView.this.normalFlow.indexOf(host).get();
+                }
 
                 while (true) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)